fil-ast/src/expr.rs

```diff
@@ -25,50 +190,55 @@ impl std::fmt::Display for Op {
 
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd)]
 /// A unary uninterpreted function over integers.
-pub enum UnFn {
-    /// The `pow2` function
+pub enum Fn {
```

Comment:
`ast::Fn` is currently used in a lot of `ir` locations, but we might want to create an `ir::Fn` version that simplifies a lot of the functions away. For example, `Log2` and `Pow2` can be replaced by `log(2, ...)` and `pow(2, ...)`, and similarly `CLog` can be replaced with `ceil(log(...))` which could be done in `astconv`.